connection_test.go

```diff
@@ -1322,3 +1322,28 @@ func TestInvalidTransportHeaders(t *testing.T) {
 		})
 	}
 }
+
+func TestCustomDialer(t *testing.T) {
+	sopts := testutils.NewOpts().NoRelay()
+	testutils.WithTestServer(t, sopts, func(t testing.TB, ts *testutils.TestServer) {
+		server := ts.Server()
+		testutils.RegisterEcho(server, nil)
+		customDialerCalled := false
+
+		copts := testutils.NewOpts().SetDialer(func(ctx context.Context, hostPort string) (net.Conn, error) {
+			customDialerCalled = true
+			d := net.Dialer{}
+			return d.DialContext(ctx, "tcp", hostPort)
+		})
+
+		// Induce the creation of a connection from client to server.
+		client := ts.NewClient(copts)
+		require.NoError(t, testutils.CallEcho(client, ts.HostPort(), ts.ServiceName(), nil))
+		assert.True(t, customDialerCalled, "custom dialer used for establishing connection")
+
+		// Re-use
+		testutils.AssertEcho(t, client, ts.HostPort(), ts.ServiceName())
+
+		client.Close()
```

Comment:
nit: no need, the `WithTestServer` will automatically close clients

Reply:
removed.